pkg/resources/streamlit_acceptance_test.go

```diff
@@ -0,0 +1,430 @@
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.root_location", rootLocationWithCatalog),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.main_file", "foo"),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.query_warehouse", warehouse.ID().Name()),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.url_id"),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.default_packages"),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.user_packages.#"),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.import_urls.#"),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.external_access_integrations.#", "1"),
+					resource.TestCheckResourceAttr("snowflake_streamlit.test", "describe_output.0.external_access_integrations.0", externalAccessIntegrationId.Name()),
+					resource.TestCheckResourceAttrSet("snowflake_streamlit.test", "describe_output.0.external_access_secrets")),
+			},
+			// import - complete
+			{
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_Streamlit/complete"),
+				ConfigVariables: m(true),
+				ResourceName:    "snowflake_streamlit.test",
+				ImportState:     true,
+				ImportStateCheck: importchecks.ComposeAggregateImportStateCheck(
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "name", id.Name()),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "database", databaseId.Name()),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "schema", schemaId.Name()),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "stage", stage.ID().FullyQualifiedName()),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "directory_location", "foo"),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "main_file", "foo"),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "query_warehouse", warehouse.ID().Name()),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "external_access_integrations.#", "1"),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "external_access_integrations.0", externalAccessIntegrationId.Name()),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "title", "foo"),
+					importchecks.TestCheckResourceAttrInstanceState(helpers.EncodeSnowflakeID(id), "comment", "foo")),
+			},
+			// change externally
+			{
+				PreConfig: func() {
+					acc.TestClient().Streamlit.Update(t, sdk.NewAlterStreamlitRequest(id).WithSet(
```

Comment:
let's add one more test with external change to a non-required attr please

Reply:
Done.